### PR TITLE
Update Go to 1.21

### PR DESCRIPTION
**Pull request by niontive**
### Which issue this PR addresses:

Fixes: https://issues.redhat.com/browse/ARO-8998

### What this PR does / why we need it:
Update to Go 1.21 since 1.20 is out of support.

### Test plan for issue:

Unit tests. Will also deploy to INT

### Is there any documentation that needs to be updated for this PR?

Yes - see PR

### CODE REVIEW

File: .github/workflows/ci-go.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
   ci-from-docker:
     runs-on: ubuntu-latest
     container:
-      image: registry.access.redhat.com/ubi8/go-toolset:1.20.12-5
+      image: registry.access.redhat.com/ubi8/go-toolset:1.21.11-1.1720406008
```
Comment (niontive) [inline .github/workflows/ci-go.yml:18]: https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1

**Comment by bitoku** [comment]
It seems arointsvc.azurecr.io/ubi8/go-toolset is not updated and doesn't contain arointsvc.azurecr.io/ubi8/go-toolset:1.21.11-1.1720406008.

**Comment by niontive** [comment]
> It seems arointsvc.azurecr.io/ubi8/go-toolset is not updated and doesn't contain arointsvc.azurecr.io/ubi8/go-toolset:1.21.11-1.1720406008.

@cmarchese created a PR to update this: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/ARO.Build.Images/pullrequest/10473966

**Comment by kimorris27** [comment]
E2E failed with a `VMStartTimedOut` error on one of the masters. Kicking off a new E2E run.

**Review by kimorris27 — APPROVED** [review]

**Comment by hawkowl** [comment]
Cool :)

**Pull request merged**